File: app/src/main/java/cn/devifish/readme/provider/SearchProvider.java
package cn.devifish.readme.provider;

import cn.devifish.readme.entity.BookList;

/**
 * Created by zhang on 2017/3/23.
 *
 * 搜索内容提供者接口
 */

public interface SearchProvider {

    public BookList getSearchResult();

}

File: app/src/main/java/cn/devifish/readme/provider/SearchProvider.kt
package cn.devifish.readme.provider

import cn.devifish.readme.entity.data.BookData
import cn.devifish.readme.service.SearchService
import io.reactivex.Observable
import okhttp3.OkHttpClient

/**
 * Created by zhang on 2017/7/22.
 * 小说搜索内容提供
 */
class SearchProvider : BaseProvider(), SearchService {

    private val searchService : SearchService = super.retrofit.create(SearchService::class.java)

    override fun searchBooks(query: String): Observable<BookData> = searchService.searchBooks(query)

    override fun searchBooksByTitle(title: String): Observable<BookData> = searchService.searchBooksByTitle(title)

    override fun searchBooksByAuthor(author: String): Observable<BookData> = searchService.searchBooksByAuthor(author)

    override fun autoComplete(query: String): Observable<String> = searchService.autoComplete(query)

}